mlflow/langchain/databricks_dependencies.py

```diff
@@ -193,6 +191,12 @@ def _extract_dependency_list_from_lc_model(lc_model) -> Generator[Resource, None
     yield from _extract_databricks_dependencies_from_retriever(lc_model)
     yield from _extract_databricks_dependencies_from_llm(lc_model)
 
+    if Version(langchain.__version__) >= Version("0.1.0"):
+        yield from _extract_databricks_dependencies_from_tools(lc_model)
+    # Langgraph needs langcahin version 0.2.0 or higher
+    if Version(langchain.__version__) >= Version("0.2.0"):
```

Comment:
I feel like it's much safer and clearer if we create a helper function like `is_toolnode_supported` and internally it tries to import dependencies like`from langgraph.prebuilt.tool_node import ToolNode`, instead of relying on the version.